modules/files/files.go

```diff
@@ -49,6 +49,39 @@ func CopyTerraformFolderToTemp(folderPath string, tempFolderPrefix string) (stri
 	return destFolder, nil
 }
 
+// CopyTerragruntFolderToTemp creates a copy of the given folder and all its contents in a temp folder with a unique name and the given prefix.
+// Since terragrunt uses tfvars files to specify modules, they are copied to the temporary directory as well.
+// Terraform state files are excluded as well as .terragrunt-cache to avoid overwriting contents.
+func CopyTerragruntFolderToTemp(folderPath string, tempFolderPrefix string) (string, error) {
```

Comment:
As you alluded, it would be great to refactor this to a common base function (`CopyFolderToTemp`) that takes in a filter function as an additional parameter (`ignoreFilter` perhaps?). Then, this one and `CopyTerraformFolderToTemp` can just pass different `ignoreFilter`s.